crates/aptos/src/move_tool/bytecode.rs

```diff
@@ -0,0 +1,306 @@
+
+            // write to file
+            write_to_user_only_file(
+                output_file.as_path(),
+                &output_file.display().to_string(),
+                output.as_bytes(),
+            )?;
+            report.push(
+                output_file
+                    .file_name()
+                    .expect("file name")
+                    .to_string_lossy()
+                    .to_string(),
+            );
+        }
+
+        Ok(match report.len() {
+            0 => "no bytecode modules found".to_owned(),
+            1 => format!("{}/{}", last_out_dir, report[0]),
+            _ => format!("{}/{{{}}}", last_out_dir, report.into_iter().join(",")),
+        })
+    }
+
+    fn disassemble(&self, bytecode_path: &Path) -> Result<String, CliError> {
+        let bytecode_bytes = read_from_file(bytecode_path)?;
+        let move_path = bytecode_path.with_extension(MOVE_EXTENSION);
+        let source_map_path = bytecode_path.with_extension(SOURCE_MAP_EXTENSION);
+
+        let source = fs::read_to_string(move_path).ok();
+        let source_map = source_map_from_file(&source_map_path).ok();
+
+        let disassembler_options = DisassemblerOptions {
+            print_code: true,
+            only_externally_visible: false,
+            print_basic_blocks: true,
+            print_locals: true,
+        };
+        let no_loc = Spanned::unsafe_no_loc(()).loc;
+        let module: CompiledModule;
+        let script: CompiledScript;
+        let bytecode = if self.is_script {
+            script = CompiledScript::deserialize(&bytecode_bytes)
+                .context("Script blob can't be deserialized")?;
+            BinaryIndexedView::Script(&script)
+        } else {
+            module = CompiledModule::deserialize(&bytecode_bytes)
+                .context("Module blob can't be deserialized")?;
+            BinaryIndexedView::Module(&module)
+        };
+
+        let mut source_mapping = if let Some(s) = source_map {
+            SourceMapping::new(s, bytecode)
+        } else {
+            SourceMapping::new_from_view(bytecode, no_loc)
+                .context("Unable to build dummy source mapping")?
+        };
+
+        if let Some(source_code) = source {
+            source_mapping.with_source_code((bytecode_path.display().to_string(), source_code));
+        }
+
+        let mut disassembler = Disassembler::new(source_mapping, disassembler_options);
+
+        if let Some(file_path) = &self.code_coverage_path {
+            disassembler.add_coverage_map(
+                CoverageMap::from_binary_file(file_path)
+                    .map_err(|_err| {
+                        CliError::UnexpectedError("Unable to read from file_path".to_string())
+                    })?
+                    .to_unified_exec_map(),
+            );
+        }
+
+        disassembler
+            .disassemble()
+            .map_err(|_err| CliError::UnexpectedError("Unable to disassemble".to_string()))
```

Comment:
Should we return the error too?  Since it's saying unable to disassemble, but with no context.

Reply:
Done